rascal-lsp/src/main/rascal/util/LanguageServer.rsc

```diff
@@ -50,6 +51,11 @@ alias LensDetector     = rel[loc src, Command lens] (Tree /*input*/);
 alias CommandExecutor  = void (Command /*command*/);
 alias InlayHinter      = list[InlayHint] (Tree /*input*/);
 
+// a decorator returns the current decorations for this tree,
+// per invocation you have to repeat all styling for this file,
+// otherwise they are cleared.
+alias Decorator        = map[str baseStyle, lrel[loc range, Maybe[MarkdownString] hoverMessage, set[InstanceDecoration] extraDecorations]] (Tree /*input*/);
```

Comment:
does this mean that the strings have to be in the order of appearance in the file? Otherwise better to use `rel`. If this is required, then runtime asserts in debug mode should help the user finding issues

Reply:
can the Maybe be avoided?